### PR TITLE
chore: initialization project

### PR DESCRIPTION
**Pull request by helen-frank**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
/cc @jwcesign @Vacant2333 @IRONICBo This is the beginning

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: .gitignore copy [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Go toolchain
```
Comment (jwcesign) [inline .gitignore copy:1]: Delete this file.

File: charts/karpenter/crds/karpenter.k8s.gcp_gkenodeclasses.yaml [hunk 2]
```diff
@@ -0,0 +1,323 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.16.4
+  name: gkenodeclasses.karpenter.k8s.gcp
```
Comment (jwcesign) [inline charts/karpenter/crds/karpenter.k8s.gcp_gkenodeclasses.yaml:7]: gcenodeclass, google compute engine

File: charts/karpenter/crds/karpenter.k8s.gcp_gkenodeclasses.yaml
Comment (jwcesign) [inline charts/karpenter/crds/karpenter.k8s.gcp_gkenodeclasses.yaml:1]: Delete this file?

**Review by jwcesign — APPROVED** [review]

File: pkg/apis/v1alpha1/gcenodeclass.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	"fmt"
+
+	"github.com/mitchellh/hashstructure/v2"
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GCENodeClassSpec is the top level specification for the GCP Karpenter Provider.
+// This will contain the configuration necessary to launch instances in GCP.
+type GCENodeClassSpec struct {
```
Comment (jwcesign) [inline pkg/apis/v1alpha1/gcenodeclass.go:29]: GCE or ce? 
https://cloud.google.com/products/compute?hl=en

What do you think?

Comment (helen-frank) [reply pkg/apis/v1alpha1/gcenodeclass.go:29]: gce

Comment (helen-frank) [reply pkg/apis/v1alpha1/gcenodeclass.go:29]: Google Compute Engine

**Pull request merged**